lib/src/main/scala/spinal/lib/CrossClock.scala

```diff
@@ -64,12 +65,16 @@ class BufferCC[T <: Data](val dataType: T, init :  => T, val bufferDepth: Option
 
   buffers(0) := io.dataIn
   buffers(0).addTag(crossClockDomain)
+  inputAttributes.foreach(buffers(0).addTag)
   for (i <- 1 until finalBufferDepth) {
     buffers(i) := buffers(i - 1)
     buffers(i).addTag(crossClockBuffer)
   }
+  buffers.map(b => allBufAttributes.foreach(b.addTag))
 
   io.dataOut := buffers.last
+
+  addAttribute("keep_hierarchy", "TRUE")
```

Comment:
I think we only need a `keep_hierarchy` here, since what I'm trying to do is prevent flattening to get a predictable port name.  I'm not sure if the other tools have the same semantics.